### PR TITLE
BUG/ENH: fix pyarrow quantile xfails

### PR DESCRIPTION
**Pull request by jbrockmendel**
fix 156 xfails and shave 49 seconds off the test runtime

### CODE REVIEW

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -1337,6 +1337,28 @@ def test_quantile(data, interpolation, quantile, request):
     data = data.take([0, 0, 0])
     ser = pd.Series(data)
     result = ser.quantile(q=quantile, interpolation=interpolation)
+
+    if pa.types.is_timestamp(pa_dtype) and interpolation not in ["lower", "higher"]:
+        # rounding error will make the check below fail
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:1342]: For the rounding errors for `interpolation not in ["lower", "higher"]`, is it due to the int casting or just the quantile computation?

Comment (jbrockmendel) [reply pandas/tests/extension/test_arrow.py:1342]: not sure where it comes in.  just checked that the results match what we get with DTA

**Comment by jbrockmendel** [comment]
@mroeschke gentle ping on pyarrow xfail PRs

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**